dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -220,6 +237,58 @@ function Catalog(props: ICatalogProps) {
         intersection(filters[filterNames.CATEGORY], getOperatorCategories(c)).length,
     );
 
+  // avoid repeating the same requests with the same params with useCallback
+  // avoid overloading the backend with a max rate of 2 reqs/second (= debounce 0.5s)
+  const debouncedFetchCharts = useCallback(
+    debounce(() => {
+      fetchCharts(cluster, namespace, reposFilter, page, size, records, searchFilter);
+    }, 500),
```

Comment:
As far as I could check, the denounce is working accordingly working with the specified time (1s in the gif):

![debounce](https://user-images.githubusercontent.com/11535726/105200842-89a10d80-5b40-11eb-8857-beba7c3ef6c8.gif)

Other if we remove the `debounce` and the `useCallback` each request is being duplicated since we wouldn't store the returned values of the function. Here's an example:

![nodebounce](https://user-images.githubusercontent.com/11535726/105200752-742be380-5b40-11eb-979b-ff8358f697c6.gif)

Reply:
Nice, thanks for double-checking this. You will need to add some tests for this though. Also, I think that the usage of `debounce` is legitimate but `useCallback` seems a hack for fixing the problem of not re-requesting the same page (in theory we should be taking the count of the pages we have already requested, and in fact we are, and use that to decide to trigger or not the function).

In any case, I am fine with this as long as you ensure it with a test.